### ISSUE TITLE
Replace legacy layer types with nodes

### ISSUE THREAD

**Issue by Keavon** [GraphiteEditor/Graphite#768]
**Task: Switch LayerDataType struct values to nodes**

```rs
#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
/// Represents different types of layers.
pub enum LayerDataType {
	/// A layer that wraps a [FolderLayer] struct.
	Folder(FolderLayer),
	/// A layer that wraps a [ShapeLayer] struct.
	Shape(ShapeLayer),
	/// A layer that wraps a [TextLayer] struct.
	Text(TextLayer),
	/// A layer that wraps an [ImageLayer] struct.
	Image(ImageLayer),
        Imaginate(ImaginateLayer),
}
```
~~`FolderLayer`~~, `ShapeLayer`, `TextLayer`, and `ImageLayer` live in `/graphene`. Each of these structs (saving `FolderLayer` for another issue) should be converted into nodes using the new node system in `/node-graph`.

---

*Questions for @TrueDoctor:*
- *How does this approach relate to the separation between "document graph" and "proto graph"?*
- *Can we continue to reference the other data structures defined in `/graphene` which are used by those three (four) structs once they are ported to nodes in `/node-graph`?*
- *Where do these new vector-editing-specific nodes go in the node graph folder structure?*
- *Any other things to be aware of?*

**Comment by TrueDoctor** [comment]
These would be document graph nodes so we are kinda blocked by my work on document nodes there

**Comment by Keavon** [comment]
@TrueDoctor Ok, thanks for clarifying that. Would you be able to meet with @0HyperCube and me soon to design that?
